pkg/reloader/reloader.go

```diff
@@ -218,9 +218,15 @@ func (r *Reloader) Watch(ctx context.Context) error {
 		}
 
 		if err := r.apply(ctx); err != nil {
-			// Critical error.
-			// TODO(bwplotka): There is no need to get process down in this case and decrease availability, handle the error in different way.
-			return err
+			switch err := errors.Cause(err).(type) {
+			case *os.PathError:
+				r.watchErrors.Inc()
+				level.Error(r.logger).Log("msg", "path error", "err", err)
+			default:
+				// Critical error.
+				// TODO(bwplotka): There is no need to get process down in this case and decrease availability, handle the error in different way.
```

Comment:
Maybe we should still address this as well. There's no need to get the process down. Maybe decision can be made by the caller side.